src/API/Google/MerchantReport.php

```diff
@@ -84,7 +84,9 @@ public function get_product_view_report( $next_page_token = null ): array {
 			->set_client( $this->service, $this->options->get_merchant_id() )
 			->get_results();
 
-			foreach ( $response->getResults() as $row ) {
+			$results = $response->getResults() ?? [];
+
+			foreach ( $results as $row ) {
```

Comment:
I'm a bit confused about this part of the code.  First, we do get the results from the query and save it to he response with `get_results()`: 

```php
$response = $query
			->set_client( $this->service, $this->options->get_merchant_id() )
			->get_results();
```
Then we call `$response->getResults()` 

So? Are we getting the results of the results?

Reply:
Yeah, I get that it's a bit confusing, but I think that's just how the Google Client works. It fetches the results from the queries first and then the rows. Maybe it would be clearer if the Google client had methods like `get_results` and `get_rows`.

See a similar example here (line 162 and 166):

https://github.com/woocommerce/google-listings-and-ads/blob/71b864b49f0740b9a7e936a0c1d3fcad5708488c/src/API/Google/MerchantReport.php#L160-L168